Create log file tests

amankumar1 opened the pull request:
Currently, log files are not being generated in the testing
framework.

Three new tests have been added to check the following:

*Log files are generated when the JIT is run with the right options.
*The log files generated have content.
*The log files have certain keywords present (see LogFileTest.cpp).

More types of logs and keywords to look for in a particular type of log
can be added in the last test.

Signed-off-by: Aman Kumar <amank@ca.ibm.com>

fjeremic commented:
Excellent documentation. We should all take note and strive for this level of detail in our commits.

0xdaryl commented on fvtest/compilertest/tests/LogFileTest.cpp:
```diff
@@ -0,0 +1,262 @@
+bool
+TestCompiler::LogFileTest::fileExists(const char *name)
+   {
+   // TODO: use port library for file operations
```
Good idea.  I'd rather see that done now than hidden behind a TODO that will never be addressed.  Granted, the LimitFileTest.cpp has the same TODO, but arguably that one should be addressed sooner rather than later (but not in this pull request).

However, you are already using C++ streams for file I/O (and for good reason) throughout this test.  An alternative that I'm not opposed to is to make consistent use of C++ streams for all I/O operations throughout this test case.

0xdaryl commented on fvtest/compilertest/tests/LogFileTest.cpp:
```diff
@@ -0,0 +1,262 @@
+
+TestCompiler::LogFileTest::LogFileTest()
+   {
+   // Don't use fork(), since that doesn't let us initialize the JIT
```
In general, please replace uses of `JIT` with `compiler` throughout this file and the commit message.

amankumar1 commented:
All requested changes incorporated:
* Replaced JIT with compiler
* Removed the `//unaryTest.invokeUnaryTests();` line
* Using C++ streams for all I/O operations in this test

0xdaryl commented:
Thanks for the changes Aman.  They look fine.  I do have a minor issue with your commit message, however, before merging.

Can you add a space between the `*` bullet and the following text so that it formats properly in Markdown syntax?  e.g.,  `* Log files...` instead of `*Log files`  Also, please change `JIT` to `compiler`.

amankumar1 commented:
Commit message updated.
